internal/engine/wazevo/engine.go

```diff
@@ -264,17 +257,48 @@ func (e *engine) compileModule(ctx context.Context, module *wasm.Module, listene
 		// At this point, relocation offsets are relative to the start of the function body,
 		// so we adjust it to the start of the executable.
 		for _, r := range rels {
+			r.Caller = fref
 			r.Offset += int64(totalSize)
 			e.rels = append(e.rels, r)
 		}
 
 		bodies[i] = body
-		totalSize += len(body)
+		totalSize += len(body) + e.machine.RelocationTrampolineSize(rels)
```

Comment:
I think the logic is broken here. Inside RelocationTrampolineSize, it checks `rels[i].TrampolineOffset > 0` but TrampolineOffset is always zero at this point. In fact
```diff
--- a/internal/engine/wazevo/engine.go
+++ b/internal/engine/wazevo/engine.go
@@ -263,7 +263,11 @@ func (e *engine) compileModule(ctx context.Context, module *wasm.Module, listene
                }
 
                bodies[i] = body
-               totalSize += len(body) + e.machine.RelocationTrampolineSize(rels)
+               s := e.machine.RelocationTrampolineSize(rels)
+               if s != 0 {
+                       panic("BUG")
+               }
+               totalSize += len(body) + s
                if wazevoapi.PrintMachineCodeHexPerFunction {
                        fmt.Printf("[[[machine code for %s]]]\n%s\n\n", wazevoapi.GetCurrentFunctionName(ctx), hex.EncodeToString(body))
                }
```

this obvious assertion doesn't get hit after running `wazero run pg.wasm`.

Reply:
This indicates that the current code is not starting with the worst case size, but instead "smallest size" scenario.